### ISSUE TITLE
[Paper Note] Data-efficient pre-training by scaling synthetic megadocs, Konwoo Kim+, arXiv'26, 2026.03

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#4981]
# URL
- https://arxiv.org/abs/2603.18534
# Authors
  - Konwoo Kim
  - Suhas Kotha
  - Yejin Choi
  - Tatsunori Hashimoto
  - Nick Haber
  - Percy Liang
# Abstract
  - Synthetic data augmentation has emerged as a promising solution when pre-training is constrained by data rather than compute. We study how to design synthetic data algorithms that achieve better loss scaling: not only lowering loss at finite compute but especially as compute approaches infinity. We first show that pre-training on web data mixed with synthetically generated rephrases improves i.i.d. validation loss on the web data, despite the synthetic data coming from an entirely different distribution. With optimal mixing and epoching, loss and benchmark accuracy improve without overfitting as the number of synthetic generations grows, plateauing near $1.48\times$ data efficiency at 32 rephrases per document. We find even better loss scaling under a new perspective: synthetic generations from the same document can form a single substantially longer megadocument instead of many short documents. We show two ways to construct megadocs: stitching synthetic rephrases from the same web document or stretching a document by inserting rationales. Both methods improve i.i.d. loss, downstream benchmarks, and especially long-context loss relative to simple rephrasing, increasing data efficiency from $1.48\times$ to $1.80\times$ at $32$ generations per document. Importantly, the improvement of megadocs over simple rephrasing widens as more synthetic data is generated. Our results show how to design synthetic data algorithms that benefit more from increasing compute when data-constrained.
# Translation (by gpt-5-nano)
- 合成データ拡張は、データ量によって事前学習が制約される場合に有望な解として浮上してきた。私たちは、有限の計算資源の下で損失を低減するだけでなく、計算資源が無限大に近づくときに特に損失スケーリングを改善するような合成データアルゴリズムを設計する方法を検討する。まず、ウェブデータを合成的再表現（言い換え）で生成したデータと混合して事前学習を行うと、合成データが全く異なる分布から来ているにもかかわらず、ウェブデータ上の独立同分布(i.i.d.)検証損失が改善されることを示す。最適な混合とエポック設定により、損失とベンチマーク精度は、合成生成の数が増えるにつれて過学習を避けつつ改善し、文書あたり32回の再表現でデータ効率は約1.48倍で頭打ちする。

さらに新たな視点では、より良い損失スケーリングが得られる。同じ文書から得られる合成再表現を用いた生成は、多数の短い文書を作るのではなく、ひとつの長大なメガ長文を形成することができる。メガ長文を構築する二つの方法を示す。1) 同じウェブ文書からの合成再表現をつなぎ合わせること、2) 根拠を挿入して文書を伸長させること。どちらの方法も、単純な再表現と比較してi.i.d.損失、下流ベンチマーク、特に長文脈での損失を改善し、文書あたり32回の生成でデータ効率を1.48倍から1.80倍へ向上させる。重要なのは、メガ長文による改善は、より多くの合成データが生成されるほど、単純な再表現よりも大きくなるという点である。我々の結果は、データが制約されている状況で、計算資源を増やすほど恩恵を受けるような合成データアルゴリズムを設計する方法を示している。
# Summary (by gpt-4o-mini)
- 合成データ拡張は、限られたデータでの事前学習に有効である。この研究では、有限の計算資源下での損失低減や、無限大に近づくときの損失スケーリングの改善を目指す。合成的再表現との混合で事前学習した場合、異なる分布からの合成データでもi.i.d.検証損失が改善され、データ効率は約1.48倍で頭打ちとなる。新たなアプローチとして、同文書からの合成再表現を用い、短文の代わりに長大なメガ長文を形成する手法を提案。これにより、損失とベンチマークの改善が見られ、データ効率は1.80倍に向上。合成データ生成が増えるほど、メガ長文による効果も増大することが示された。

**Comment by AkihikoWatanabe** [comment]
元ポスト:https://x.com/rosinality/status/2034924371563675806?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
著者ポスト:https://x.com/konwookim/status/2035029597491011984?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
著者ポスト:https://x.com/percyliang/status/2035112178580398341?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
- データよりもコンピューティングリソースのスケーリングの方が早く進んでおり、データ効率の高い事前学習レシピが重要となっている
- 事前学習において、合成データがi.i.d.なwebデータの損失減らすことに寄与するかを調査
- 300Mモデルで200M tokenを学習した際にどれだけi.i.d.なwebデータのlossを低減させられるかを調査
- 最初に最もシンプルなdata augmentationであるrephrasingを調査したところ、文書単位でのrephrasingの回数が増えるにつれて、web lossとdownstreamベンチマークでのエラー率が単調に改善
- 続いて、ある文書をrephraseした文書を結合することで、単一の大きな文書（=megadoc）を構成する手法を提案し、megadocを利用することでさらにlossが改善することを確認。megadocの構成方法として下記三種類を提案し:
  - Real First Stitched: `文書に対するrephraseをG個生成し、それらを結合することでmegadocを構成する手法。実データを結合の頭にもってくる。
  - Real Last Stitched: Real First Stichedと同様の処理をするが、実データを結合の末尾に持ってくる手法
  - Latent Thoughts: 文書をG+1個の同じ長さのピースに分割し、ピース間を埋めるrationaleを合成して結合する手法。rationaleは<think></think>タグで囲う。
- Real First Stitched と Real Last Stitched を比較したところ、後者の方が性能が良かった。
  - 後者の方が性能が良い考察として、epiplexity #4145 の観点から考察をしている。前提として実文書の方が複雑で情報量が多いと考えたときに、Real First Stitched の場合は実文書の情報からrephraseを学ぶという簡単な変換（生成）を実施すればよいのに対し、Real Last Stitchedの場合逆で、rephraseからより詳細で複雑な実文書に変換（生成）するというタスクを実施せねばならない。このため、後者の方がより計算的に困難な関数を学習する必要があり（すなわち、epiplexityが高い学習設定ということ; epiplexityが高い学習設定の方がモデルの汎化性能が高くなる）、学習の結果より高い汎化性能を獲得しているのではないか、と考察している。

<img width="1048" height="852" alt="Image" src="https://github.com/user-attachments/assets/2eb9eab2-b9a1-4e87-838b-af1d16adfe11" />

- また、モデルをアンサンブルした場合の性質についても考察がされており、self-distillationは単体モデルの性能を向上させることに寄与するが、アンサンブルするモデルの数を増やすと実データを用いたモデルと最終的には性能が同等となることが予測され、達成可能なピーク性能がアンサンブルによってブーストされる効果は観測できなかった。一方で、Rephrasingによる合成データによって学習されたモデルはアンサンブルによって達成可能な性能のピーク値がブーストされると考えられる。
<img width="1016" height="785" alt="Image" src="https://github.com/user-attachments/assets/9269fbbd-414c-490d-99f4-d85f89613f3d" />

**Comment by AkihikoWatanabe** [comment]
関連:
- #1937
